### PR TITLE
Add counterpartyChannelId param to IBCModule.OnChanOpenAck

### PR DESCRIPTION
**Pull request by catShaark**
## Description
ref: #1075

I add counterpartyChannelID as a param to IBCModule.OnChanOpenAck to support ibc module that use counterpartyChannelID in their OnChanOpenAck logic. #1075 

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by faddat** [comment]
@joe-bowman -- is this a good solution for supporting cosmwasm with ibc-go v3 and cosmos-sdk 46?

File: modules/core/05-port/types/module.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type IBCModule interface {
 		portID,
 		channelID string,
 		counterpartyVersion string,
+		counterpartyChannelID string,
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:55]: could you change the ordering such that `counterpartyChannelID` comes before `counterpartyVersion`? See [proposed spec changes](https://github.com/cosmos/ibc/pull/679)?

**Review by colin-axner — COMMENTED** [review]
Thanks for the contribution! 

This needs a changelog entry under the API breaking section and an addition to the `docs/v2-to-v3.md` migrations docs, but I can add both in a followup pr if you feel uncertain about adding that

File: modules/core/05-port/types/module.go (hunk 1 above)
Comment (catShaark) [reply modules/core/05-port/types/module.go:55]: yes, I'll do it

Comment (catShaark) [reply modules/core/05-port/types/module.go:55]: > Thanks for the contribution!
> 
> This needs a changelog entry under the API breaking section and an addition to the `docs/v2-to-v3.md` migrations docs, but I can add both in a followup pr if you feel uncertain about adding that

I can make the pr

Comment (catShaark) [reply modules/core/05-port/types/module.go:55]: Can I also make pr to the spec, I really want to make contribution to the spec

**Comment by colin-axner** [comment]
issue closed by #1086

**Pull request closed without merging**